bee-protocol/src/worker/requester/transaction.rs

```diff
@@ -69,23 +87,47 @@ impl<'a> TransactionRequesterWorker<'a> {
 
             if let Some(peer) = Protocol::get().peer_manager.handshaked_peers.get(epid) {
                 if index > peer.snapshot_milestone_index() && index <= peer.last_solid_milestone_index() {
-                    Protocol::get().requested_transactions.insert(hash, index);
                     SenderWorker::<TransactionRequest>::send(
                         epid,
                         TransactionRequest::new(cast_slice(hash.as_trits().encode::<T5B1Buf>().as_i8_slice())),
                     );
-                    break;
+                    return true;
                 }
             }
         }
+        false
+    }
+
+    async fn retry_requests(&mut self) {
+        let mut retry_counts = 0;
+
+        for mut transaction in Protocol::get().requested_transactions.iter_mut() {
+            let (hash, (index, instant)) = transaction.pair_mut();
+            let now = Instant::now();
+            if (now - *instant).as_secs() > RETRY_INTERVAL_SECS {
+                debug!("Transaction timed out, retrying request.");
+                if self.process_request_unchecked(hash.clone(), *index).await {
+                    *instant = now;
+                }
+                retry_counts += 1;
```

Comment:
Should the retry count be incremented in the if ?

Reply:
well I suppose it should